README.md
primofree
=========

Primo Free Theme


changelog.txt
Primo Changelog

****************************
Version 1.4 (10-03-17)
- Fixed header already sent warning

Version 1.3 (07-21-17)
- Added visible edit links in the customizer preview

Version 1.2 (04-20-17)
- Fixed issue with PHP 7.1
- Fixed issue with loading google fonts (used protocol relative link)

Version 1.1 (10-06-16)
- Fixed display issue for dropdown and searchbar in sidebar.

Version 1.0.37 (04-22-16)
- Site font issue fixed

Version 1.0.36 (11-26-15)
- Fix few minor issues related to customization

Version 1.0.35 (16-07-15)
- Change in theme customizer as per wordpress.org guidelines

Version 1.0.34 (1-23-15)
- Fixed conflict between slider free and pro plugin.
- Added new plugin suggestion.

Version 1.0.33 (11-25-14)
- Removed theme activation notice.
- Changed default links for slider.

Version 1.0.32 (11-16-14)
- Removed cyberchimps_mlu_init() as it has register_post_type() which is plugin-territory functionality.
- Added sanitization to customizer options.
- Removed deprecated screen_icon().

Version 1.0.31 (9-24-14)
- Placeholder fix for IE8/9.

Version 1.0.30 (9-08-14)
- Added iFeature slider to plugin suggestion list.

Version 1.0.29 (8-22-14)
- Updated Translations.

Version 1.0.28 (8-08-14)
- Fixed conflict of footer widgets with other widgets.
- Responsive fix for uploaded video.

Version 1.0.27 (7-25-14)
- Layout fixed for multiple rows of footer widget.

Version 1.0.26 (7-18-14)
- Fixed broken link to CyberChimps logo.

Version 1.0.25 (7-11-14)
- Replaced logo with text in footer credit link.

Version 1.0.24 (7-04-14)
- Added option for Instagram Icon.

Version 1.0.23 (6-27-14)
- Core Translation Updated.

Version 1.0.22 (6-20-14)
- Code reformatting: Changed spaces to tabs.

Version 1.0.21 (6-13-14)
- Code reformatted: Changed spaces to tabs.

Version 1.0.20 (6-06-14)
- Uploading due to technical problem.

Version 1.0.19 (6-06-14)
- Code reformatting: Changed spaces to tabs.

Version 1.0.18 (5-30-14)
- Fixed casing of CyberChimpsParallax class.

Version 1.0.17 (5-23-14)
- Updated ugrade text.
- Removed left sidebar from widget section.
- Removed "Custom Callout" from sub settings.

Version 1.0.16 (5-16-14)
- Correct spelling of CyberChimps/CSS.
- Change text for Google font field.
- Added cyberchimps_ prefix to couple of functions.
- Changed CyberChimps logo on theme options page.
- Changed "alt" text of logo to the site name instead of "logo".
- Regenerated ".mo" files as they were 0KB each before, hence breaking the translation.

Version 1.0.15 (5-9-14)
- Fixed W3C validator error issue with IE compatibility meta tags.
- Fixed missing image error: removed unused function.
- Added FooBox as recommended plugin.

Version 1.0.14 (5-2-14)
- Fixed byline toggle issue.

Version 1.0.13 (4-24-14)
- Added recommendation for clef plugin.
- Updated translations.
- Youtube video fixed.
- IE 8/9 fix for Menu, Elements and Footer Background.

Version 1.0.12 (4-16-14)
- Changed social font icons to font awesome.
- Removed unwanted function_exists() conditional wrapper.

Version 1.0.11 (4-9-14)
-Added upgrade button to the free theme customizer.
-Fixed Vimeo video responsiveness.

Version 1.0.10 (4-3-14)
- Added new background option.
- Removed wp_register_style(or script) for local css/js files.

Version 1.0.9 (3-27-14)
- Updated Elements Translation.

Version 1.0.8 (3-21-14)
- Fixed IE8 compatibility issue.

Version 1.0.7 (3-12-14)
- Fixed duplicate plugin install notification in option page.
- Modified "More Themes" page to show up all CyberChimps free theme with option to install/activate right there.

Version 1.0.6 (3-5-14)
- Fixed conflict between blog post excerpt option and bbpress user profile page.
- Set up element options defaults for better wp.org preview.
- Added filter for background default args.
- HTML Error fix: Added space before menu>li>attribute.

Version 1.0.5 (2-27-14)
- Updated elements images along with boxes images.
- Fixed Read more text option.
- Responsive toggle fix: tweaked full-width-container width.
- Background option fix.
- Removed post format icon from page.
- Updated font and screenshot.
- Fixed word wrap.

Version 1.0.4 (2-20-14)
- Theme URI removed

Version 1.0.3 (2-18-14)
- Slider is made to be start sliding on page load.

Version 1.0.2 (02/12/14)

- Updated translations

Version 1.0.1 (02/07/14)

- Initial bug fixes

Version 1.0.0 (01/28/14)
- Initial release.


functions.php
<?php
/**
 * Title: Function
 *
 * Description: Defines theme specific functions including actions and filters.
 *
 * Please do not edit this file. This file is part of the Cyber Chimps Framework and all modifications
 * should be made in a child theme.
 *
 * @category Cyber Chimps Framework
 * @package  Framework
 * @since    1.0
 * @author   CyberChimps
 * @license  http://www.opensource.org/licenses/gpl-license.php GPL v3.0 (or later)
 * @link     http://www.cyberchimps.com/
 */

// Load text domain.
function cyberchimps_text_domain() {
	load_theme_textdomain( 'primo', get_template_directory() . '/inc/languages' );
	add_theme_support( 'title-tag' );
}
add_action( 'after_setup_theme', 'cyberchimps_text_domain' );

// Load Core
require_once( get_template_directory() . '/cyberchimps/init.php' );
require( get_template_directory() . '/inc/admin-about.php' );

// Set the content width based on the theme's design and stylesheet.
if( !isset( $content_width ) ) {
	$content_width = 640;
} /* pixels */

// Define site info
function cyberchimps_add_site_info() {
	?>
	<p>&copy; Company Name</p>
<?php
}

add_action( 'cyberchimps_site_info', 'cyberchimps_add_site_info' );

if( !function_exists( 'cyberchimps_comment' ) ) :

// Template for comments and pingbacks.
// Used as a callback by wp_list_comments() for displaying the comments.
	function cyberchimps_comment( $comment, $args, $depth ) {
		$GLOBALS['comment'] = $comment;
		switch( $comment->comment_type ) :
			case 'pingback' :
			case 'trackback' :
				?>
				<li class="post pingback">
				<p><?php _e( 'Pingback:', 'primo' ); ?> <?php comment_author_link(); ?><?php edit_comment_link( __( '(Edit)', 'primo' ), ' ' ); ?></p>
				<?php
				break;
			default :
				?>
					<li <?php comment_class(); ?> id="li-comment-<?php comment_ID(); ?>">
					<article id="comment-<?php comment_ID(); ?>" class="comment hreview">
						<footer>
							<div class="comment-author reviewer vcard">
								<?php echo get_avatar( $comment, 40 ); ?>
								<?php printf( '%s <span class="says">' . __( 'says:', 'primo' ) . '</span>', sprintf( '<cite class="fn">%s</cite>', get_comment_author_link() ) ); ?>
							</div>
							<!-- .comment-author .vcard -->
							<?php if( $comment->comment_approved == '0' ) : ?>
								<em><?php _e( 'Your comment is awaiting moderation.', 'primo' ); ?></em>
								<br/>
							<?php endif; ?>

							<div class="comment-meta commentmetadata">
								<a href="<?php echo esc_url( get_comment_link( $comment->comment_ID ) ); ?>" class="dtreviewed">
									<time pubdate datetime="<?php comment_time( 'c' ); ?>">
										<?php
										/* translators: 1: date, 2: time */
										printf( __( '%1$s at %2$s', 'primo' ), get_comment_date(), get_comment_time() ); ?>
									</time>
								</a>
								<?php edit_comment_link( __( '(Edit)', 'primo' ), ' ' );
								?>
							</div>
							<!-- .comment-meta .commentmetadata -->
						</footer>

						<div class="comment-content"><?php comment_text(); ?></div>

						<div class="reply">
							<?php comment_reply_link( array_merge( $args, array( 'depth' => $depth, 'max_depth' => $args['max_depth'] ) ) ); ?>
						</div>
						<!-- .reply -->
					</article><!-- #comment-## -->

				<?php
				break;
		endswitch;
	}
endif; // ends check for cyberchimps_comment()

// set up next and previous post links for lite themes only
function cyberchimps_next_previous_posts() {
	if( get_next_posts_link() || get_previous_posts_link() ): ?>
		<div class="more-content">
			<div class="row-fluid">
				<div class="span6 previous-post">
					<?php previous_posts_link(); ?>
				</div>
				<div class="span6 next-post">
					<?php next_posts_link(); ?>
				</div>
			</div>
		</div>
	<?php
	endif;
}

add_action( 'cyberchimps_after_content', 'cyberchimps_next_previous_posts' );

// core options customization Names and URL's
//Pro or Free
function cyberchimps_theme_check() {
	$level = 'free';

	return $level;
}

//Theme Name
function cyberchimps_options_theme_name() {
	$text = 'Primo';

	return $text;
}

//Theme Pro Name
function cyberchimps_upgrade_bar_pro_title() {
	$text = 'Primo Pro';

	return $text;
}

//Upgrade link
function cyberchimps_upgrade_bar_pro_link() {
	$url = 'http://cyberchimps.com/store/primo-pro/';

	return $url;
}

//Doc's URL
function cyberchimps_options_documentation_url() {
	$url = 'http://cyberchimps.com/guides/c/';

	return $url;
}

// Support Forum URL
function cyberchimps_options_support_forum() {
	$url = 'http://cyberchimps.com/forum/free/primo/';

	return $url;
}

add_filter( 'cyberchimps_current_theme_name', 'cyberchimps_options_theme_name', 1 );
add_filter( 'cyberchimps_upgrade_pro_title', 'cyberchimps_upgrade_bar_pro_title' );
add_filter( 'cyberchimps_upgrade_link', 'cyberchimps_upgrade_bar_pro_link' );
add_filter( 'cyberchimps_documentation', 'cyberchimps_options_documentation_url' );
add_filter( 'cyberchimps_support_forum', 'cyberchimps_options_support_forum' );

// Help Section
function cyberchimps_options_help_header() {
	$text = 'Primo';

	return $text;
}

function cyberchimps_options_help_sub_header() {
	$text = __( 'Primo Professional Responsive WordPress Theme', 'primo' );

	return $text;
}

add_filter( 'cyberchimps_help_heading', 'cyberchimps_options_help_header' );
add_filter( 'cyberchimps_help_sub_heading', 'cyberchimps_options_help_sub_header' );

//theme specific skin options in array. Must always include option default
function cyberchimps_skin_color_options( $options ) {
	// Get path of image
	$imagepath = get_template_directory_uri() . '/inc/css/skins/images/';

	$options = array(
		'default' => $imagepath . 'default.png'
	);

	return $options;
}

add_filter( 'cyberchimps_skin_color', 'cyberchimps_skin_color_options' );

// theme specific background images
function cyberchimps_background_image( $options ) {
	$imagepath = get_template_directory_uri() . '/cyberchimps/lib/images/';
	$options   = array(
		'none'  => $imagepath . 'backgrounds/thumbs/none.png',
		'noise' => $imagepath . 'backgrounds/thumbs/noise.png',
		'blue'  => $imagepath . 'backgrounds/thumbs/blue.png',
		'dark'  => $imagepath . 'backgrounds/thumbs/dark.png',
		'space' => $imagepath . 'backgrounds/thumbs/space.png'
	);

	return $options;
}

add_filter( 'cyberchimps_background_image', 'cyberchimps_background_image' );

// theme specific typography options
function cyberchimps_typography_sizes( $sizes ) {
	$sizes = array( '8', '9', '10', '12', '14', '16', '20' );

	return $sizes;
}

function cyberchimps_typography_styles( $styles ) {
	$styles = array( 'normal' => 'Normal', 'bold' => 'Bold' );

	return $styles;
}

function cyberchimps_typography_faces( $orig ) {

    $new = array(
        '"Trocchi", Times, serif'          => 'Trocchi',
        '"Quattrocento Sans", Times, serif' => 'Quattrocento'
    );
    $new = array_merge( $new, $orig );
    return $new;
}

function cyberchimps_typography_defaults() {
    $default = array(
        'size'  => '14px',
        'face'  => '"Quattrocento Sans", Times, serif',
        'style' => 'normal',
        'color' => '#555555'
    );
    return $default;
}

function cyberchimps_typography_heading_defaults() {
    $default = array(
        'size'  => '',
        'face'  => '"Trocchi", Times, serif',
        'style' => '',
        'color' => '',
    );
    return $default;
}

add_filter( 'cyberchimps_typography_sizes', 'cyberchimps_typography_sizes' );
add_filter( 'cyberchimps_typography_styles', 'cyberchimps_typography_styles' );
add_filter( 'cyberchimps_typography_faces', 'cyberchimps_typography_faces' );
add_filter( 'cyberchimps_typography_defaults', 'cyberchimps_typography_defaults' );
add_filter( 'cyberchimps_typography_heading_defaults', 'cyberchimps_typography_heading_defaults' );

// Default for twitter bar handle
function cyberchimps_twitter_handle_filter() {
	return 'WordPress';
}

add_filter( 'cyberchimps_twitter_handle_filter', 'cyberchimps_twitter_handle_filter' );

function primo_customize_edit_links( $wp_customize ) {


   $wp_customize->selective_refresh->add_partial( 'blogname', array(
'selector' => '.site-title a'
) );

	$wp_customize->selective_refresh->add_partial( 'blogdescription', array(
		'selector' => '.top-head-description'
	) );

	$wp_customize->selective_refresh->add_partial( 'cyberchimps_options[custom_logo]', array(
		'selector' => '#logo'
	) );

	$wp_customize->selective_refresh->add_partial( 'cyberchimps_options[theme_backgrounds]', array(
		'selector' => '#social'
	) );

	$wp_customize->selective_refresh->add_partial( 'cyberchimps_options[searchbar]', array(
		'selector' => '#navigation #searchform'
	) );

	$wp_customize->selective_refresh->add_partial( 'cyberchimps_options[footer_show_toggle]', array(
		'selector' => '#footer_wrapper'
	) );

	$wp_customize->selective_refresh->add_partial( 'cyberchimps_options[footer_copyright_text]', array(
		'selector' => '#copyright'
	) );

	$wp_customize->selective_refresh->add_partial( 'nav_menu_locations[primary]', array(
		'selector' => '#navigation .nav'
	) );

	$wp_customize->selective_refresh->add_partial( 'cyberchimps_options[blog_title]', array(
		'selector' => '.page-title'
	) );

	$wp_customize->selective_refresh->add_partial( 'cyberchimps_options[footer_show_toggle]', array(
	'selector' => '#footer-widget-container'
) );

}
add_action( 'customize_register', 'primo_customize_edit_links' );
add_theme_support( 'customize-selective-refresh-widgets' );

add_action( 'admin_notices', 'radiant_admin_notices' );
function radiant_admin_notices()
{
	$admin_check_screen = get_admin_page_title();

	if( !class_exists('SlideDeckPlugin') )
	{
	$plugin='slidedeck/slidedeck.php';
	$slug = 'slidedeck';
	$installed_plugins = get_plugins();

	 if ( $admin_check_screen == 'Manage Themes' || $admin_check_screen == 'Theme Options Page' )
	{
		?>
		<div class="notice notice-info is-dismissible" style="margin-top:15px;">
		<p>
			<?php if( isset( $installed_plugins[$plugin] ) )
			{
			?>
				 <a href="<?php echo admin_url( 'plugins.php' ); ?>">Activate the SlideDeck Lite plugin</a>
			 <?php
			}
			else
			{
			 ?>
			 <a href="<?php echo wp_nonce_url( self_admin_url( 'update.php?action=install-plugin&plugin=' . $slug ), 'install-plugin_' . $slug ); ?>">Install the SlideDeck Lite plugin</a>
			 <?php } ?>

		</p>
		</div>
		<?php
	}
	}

	if( !class_exists('WPForms') )
	{
	$plugin = 'wpforms-lite/wpforms.php';
	$slug = 'wpforms-lite';
	$installed_plugins = get_plugins();
	 if ( $admin_check_screen == 'Manage Themes' || $admin_check_screen == 'Theme Options Page' )
	{
		?>
		<div class="notice notice-info is-dismissible" style="margin-top:15px;">
		<p>
			<?php if( isset( $installed_plugins[$plugin] ) )
			{
			?>
				 <a href="<?php echo admin_url( 'plugins.php' ); ?>">Activate the WPForms Lite plugin</a>
			 <?php
			}
			else
			{
			 ?>
	 		 <a href="<?php echo wp_nonce_url( self_admin_url( 'update.php?action=install-plugin&plugin=' . $slug ), 'install-plugin_' . $slug ); ?>">Install the WP Forms Lite plugin</a>
			 <?php } ?>
		</p>
		</div>
		<?php
	}
	}

	if ( $admin_check_screen == 'Manage Themes' || $admin_check_screen == 'Theme Options Page' )
	{
	?>
		<div class="notice notice-success is-dismissible">
				<b><p>Liked this theme? <a href="https://wordpress.org/support/theme/primo-lite/reviews/#new-post" target="_blank">Leave us</a> a ***** rating. Thank you! </p></b>
		</div>
		<?php
	}

}
